flocker/node/agents/ebs.py

```diff
@@ -390,122 +410,221 @@ def ec2_client(region, zone, access_key_id, secret_access_key):
     :return: An ``_EC2`` giving information about EC2 client connection
         and EC2 instance zone.
     """
-    # Set 2 retry knobs in Boto to BOTO_NUM_RETRIES:
-    # 1. ``num_retries``:
-    # Request automatic exponential backoff and retry
-    # attempts by Boto if an EC2 API call fails with
-    # ``RequestLimitExceeded`` due to system load.
-    # 2. ``metadata_service_num_attempts``:
-    # Request for retry attempts by Boto to
-    # retrieve data from Metadata Service used to retrieve
-    # credentials for IAM roles on EC2 instances.
-    if not config.has_section('Boto'):
-        config.add_section('Boto')
-    config.set('Boto', 'num_retries', BOTO_NUM_RETRIES)
-    config.set('Boto', 'metadata_service_num_attempts', BOTO_NUM_RETRIES)
-
-    # Get Boto EC2 connection with ``EC2ResponseError`` logged by Eliot.
-    connection = ec2.connect_to_region(region,
-                                       aws_access_key_id=access_key_id,
-                                       aws_secret_access_key=secret_access_key)
-    return _EC2(zone=zone,
-                connection=_LoggedBotoConnection(connection=connection))
-
-
-def _boto_logged_method(method_name, original_name):
+    connection = boto3.session.Session(
+        aws_access_key_id=access_key_id,
+        aws_secret_access_key=secret_access_key
+    )
+    connection._session.set_config_variable(
```

Comment:
`max_attempts` can't be configured this way, however the default limit is 5 retries with an exponential backoff between tries. This may be sufficient, or if not, more research is needed in how to set the retry config dynamically.